www-apps/gitea/gitea-1.7.4-r1.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit golang-vcs-snapshot systemd user
+
+EGO_PN="code.gitea.io/gitea"
+KEYWORDS="~amd64 ~arm"
+
+DESCRIPTION="A painless self-hosted Git service"
+HOMEPAGE="https://gitea.io"
+SRC_URI="https://github.com/go-gitea/gitea/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="pam sqlite"
+
+CDEPEND="pam? ( sys-libs/pam )"
+DEPEND="${CDEPEND}
+	dev-go/go-bindata"
+RDEPEND="${CDEPEND}
+	dev-vcs/git"
+
+DOCS=( custom/conf/app.ini.sample CONTRIBUTING.md README.md )
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+pkg_setup() {
+	enewgroup git
+	enewuser git -1 /bin/bash /var/lib/gitea git
+}
+
+gitea_make() {
+	local my_tags=(
+		bindata
+		$(usev pam)
+		$(usex sqlite 'sqlite sqlite_unlock_notify' '')
+	)
+	local my_makeopt=(
+		DRONE_TAG=${PV}
+		TAGS="${my_tags[@]}"
+	)
+	GOPATH=${WORKDIR}/${P}:$(get_golibdir_gopath) emake "${my_makeopt[@]}" "$1"
+}
+
+src_compile() {
+	gitea_make generate
+	gitea_make build
+}
+
+src_test() {
+	gitea_make test
+}
+
+src_install() {
+	einstalldocs
+	dobin gitea
+	newconfd "${FILESDIR}"/gitea.confd-r1 gitea
+	newinitd "${FILESDIR}"/gitea.initd-r2 gitea
+	systemd_dounit "${FILESDIR}"/gitea.service
+	diropts -m0750 -o git -g git
+	keepdir /etc/gitea /var/log/gitea /var/lib/gitea /var/lib/gitea/custom /var/lib/gitea/data
+}
+
+pkg_postinst() {
+	ewarn "The default configuration path changed to /etc/gitea/app.ini"
+	ewarn "Please migrate your configuration and then you can remove the"
+	ewarn "previously used /var/lib/gitea/conf directory."
+	if [[ ! -e "${EROOT}/etc/gitea/app.ini" ]]; then
+		elog "No app.ini found, copying initial config over"
+		cp "${FILESDIR}"/app.ini "${EROOT}"/etc/gitea/ || die
+		chown git:git /etc/gitea/app.ini
```

Comment:
Would fowners work here? Add die if not. When has this check become relevant? Could it be improved with has_version? Also it can be done in earlier phases, if using doins or fowners causes trouble in pkg_postinst.

Reply:
fowners should work, chown has the disadvantage that running gitea as another user (e.g. git might be taken by gitolite already), that with every update the permission have to be set again.

Reply:
This only happens when you have a new installation. If you update, the configuration file exist and won't change ownership (except this case when we change the configuration path, but I believe it's a one time action).

Reply:
Fowners is called in src_install, so it won't work here. Adding missing || die, thanks.